### ISSUE TITLE
Collapse animation is jittery in Firefox Android (preview and live)

### ISSUE THREAD

**Issue by Noisycall** [twbs/bootstrap#30976]
As the title suggests, the collapse animation is jittery on firefox mobile, both the preview and live builds.

I have tested this on two devices and both have the same issue.

Here is a link to the [JS Bin](https://jsbin.com/kezocoqaji/1/edit?html,output)

This code is taken straight from the Bootstrap Demos, along with some lorem ipsum to simulate page content.

**Comment by XhmikosR** [comment]
Firefox for Android is, unfortunately, so bad in performance. Even their latest Beta, which has improved performance, is still not as smooth as Chromium based browsers.

That being said, right now, we transition `height .35s ease`, which is known to not have the best performance because the GPU isn't used.

I'm not sure we can do anything, but patches are welcome.

CC @twbs/css-review  in case you have any ideas.

**Comment by ffoodd** [comment]
Did anyone try to use `will-change` for collapses? It's supported by Firefox for Android: not quite sure it'd improve the result, but that may worth a try.

- [`will-change` on MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/will-change)
- [Can I Use `will-change`](https://caniuse.com/#search=will-change)

@Noisycall may you try the following pen, please: https://codepen.io/ffoodd/pen/NWxGEQQ ?

**Comment by XhmikosR** [comment]
I don't want to use `wll-change` TBH. I'd rather use one of the properties that work pretty well when transitioned.

**Comment by N8n5h** [comment]
> may you try the following pen, please: https://codepen.io/ffoodd/pen/NWxGEQQ ?

I'm not OP, but I have the same problem and the performance still is a bit jittery even with `will-change`

> That being said, right now, we transition height .35s ease, which is known to not have the best performance because the GPU isn't used.

Maybe using transforms? https://medium.com/outsystems-experts/how-to-achieve-60-fps-animations-with-css3-db7b98610108

**Comment by XhmikosR** [comment]
> Maybe using transforms? https://medium.com/outsystems-experts/how-to-achieve-60-fps-animations-with-css3-db7b98610108

Well, yeah, that's what I mean with properties that work well when transitioned :)

**Comment by Noisycall** [comment]
Hello, sorry I haven't checked this thread in a while.

I have used transform and transition's before, but I'm not sure how I would apply that to the inbuilt bootstrap implemention.

> 
> 
> > Maybe using transforms? https://medium.com/outsystems-experts/how-to-achieve-60-fps-animations-with-css3-db7b98610108
> 
> Well, yeah, that's what I mean with properties that work well when transitioned :)

**Comment by XhmikosR** [comment]
Yeah, it's not so easy to make the change but we should definitely look into it.

**Comment by XhmikosR** [comment]
This might help: https://css-tricks.com/performant-expandable-animations-building-keyframes-on-the-fly/
